src/object/object_representation.c

```diff
@@ -1352,23 +1355,95 @@ or_put_varchar_internal (OR_BUF * buf, char *string, int charlen, int align)
   else
     {
       rc = or_put_byte (buf, 0xFF);
-
-      if (rc == NO_ERROR)
-	{
-	  OR_PUT_INT (&net_charlen, charlen);
-	  rc = or_put_data (buf, (char *) &net_charlen, OR_INT_SIZE);
-	}
+      compressable = true;
     }
   if (rc != NO_ERROR)
     {
       return rc;
     }
 
-  /* store the string bytes */
-  rc = or_put_data (buf, string, charlen);
-  if (rc != NO_ERROR)
+  if (compressable == true)
     {
-      return rc;
+      /* Future optimization : use a preallocated object for wrkmem in thread_entry. */
+      /* Alloc memory */
+      wrkmem = (lzo_voidp) malloc (LZO1X_1_MEM_COMPRESS);
+      if (wrkmem == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, (size_t) LZO1X_1_MEM_COMPRESS);
+	  goto cleanup;
+	}
+
+      /* Alloc memory for the compressed string */
+      /* Worst case LZO compression size from their FAQ */
+      compressed_string = (char *) malloc ((charlen + (charlen / 16) + 64 + 3) * sizeof (char));
+      if (compressed_string == NULL)
+	{
+	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY,
+		  1, (size_t) (charlen + (charlen / 16) + 64 + 3));
+	  goto cleanup;
+	}
+
+      /* Compress the string */
+      rc = lzo1x_1_compress (string, charlen, compressed_string, &compressed_length, wrkmem);
+      if (rc != NO_ERROR)
+	{
+	  goto cleanup;
```

Comment:
@lucianlazar2211 

`NO_ERROR` should be `LZO_E_OK`. 

Please also set `ER_IO_LZO_COMPRESS_FAIL` error.